syncto/tests/test_utils.py

```diff
@@ -0,0 +1,84 @@
+import os
+import unittest
+
+from binascii import hexlify
+from six import text_type, binary_type
+from uuid import uuid4, UUID
+
+from syncto.utils import (
+    bytes_to_uuid4, UUID4_VALIDATOR, uuid4_to_bytes, base64url_encode,
+    base64_to_uuid4, uuid4_to_base64
+)
+
+
+def generate_random_uuid4():
+    return text_type(uuid4())
+
+
+def generate_made_up_uuid4():
+    uuid_bytes = list(uuid4().bytes)
+    uuid_bytes[6] = binary_type('\x40')
+    uuid_bytes[8] = binary_type('\x80')
+    uuid = UUID(hexlify(binary_type(''.join(uuid_bytes))))
+    return text_type(uuid)
+
+
+class Bytes2UUID4Test(unittest.TestCase):
+    def test_a_wrong_uuid4_array_fails(self):
+        try:
+            bytes_to_uuid4(os.urandom(16))
+        except ValueError:
+            pass
+        else:
+            self.fail("16 random bytes shouldn't make a valid UUID4")
```

Comment:
Make predictive tests, they are cases where the test can fail even if it is really unlikely.